%  LaTeX support: latex@mdpi.com 
%  For support, please attach all files needed for compiling as well as the log file, and specify your operating system, LaTeX version, and LaTeX editor.

%=================================================================
\documentclass[journal,article,submit,pdftex,moreauthors]{Definitions/mdpi} 
%\documentclass[preprints,article,submit,pdftex,moreauthors]{Definitions/mdpi} 
% For posting an early version of this manuscript as a preprint, you may use "preprints" as the journal. Changing "submit" to "accept" before posting will remove line numbers.

%--------------------
% Class Options:
%--------------------
%----------
% journal
%----------
% Choose between the following MDPI journals:
% accountaudit, acoustics, actuators, addictions, adhesives, admsci, adolescents, aerobiology, aerospace, agriculture, agriengineering, agrochemicals, agronomy, ai, air, algorithms, allergies, alloys, amh, analytica, analytics, anatomia, anesthres, animals, antibiotics, antibodies, antioxidants, applbiosci, appliedchem, appliedmath, appliedphys, applmech, applmicrobiol, applnano, applsci, aquacj, architecture, arm, arthropoda, arts, asc, asi, astronomy, atmosphere, atoms, audiolres, automation, axioms, bacteria, batteries, bdcc, behavsci, beverages, biochem, bioengineering, biologics, biology, biomass, biomechanics, biomed, biomedicines, biomedinformatics, biomimetics, biomolecules, biophysica, biosensors, biosphere, biotech, birds, blockchains, bloods, blsf, brainsci, breath, buildings, businesses, cancers, carbon, cardiogenetics, catalysts, cells, ceramics, challenges, chemengineering, chemistry, chemosensors, chemproc, children, chips, cimb, civileng, cleantechnol, climate, clinbioenerg, clinpract, clockssleep, cmd, cmtr, coasts, coatings, colloids, colorants, commodities, complications, compounds, computation, computers, condensedmatter, conservation, constrmater, cosmetics, covid, crops, cryo, cryptography, crystals, csmf, ctn, curroncol, cyber, dairy, data, ddc, dentistry, dermato, dermatopathology, designs, devices, diabetology, diagnostics, dietetics, digital, disabilities, diseases, diversity, dna, drones, dynamics, earth, ebj, ecm, ecologies, econometrics, economies, education, eesp, ejihpe, electricity, electrochem, electronicmat, electronics, encyclopedia, endocrines, energies, eng, engproc, ent, entomology, entropy, environments, epidemiologia, epigenomes, esa, est, famsci, fermentation, fibers, fintech, fire, fishes, fluids, foods, forecasting, forensicsci, forests, fossstud, foundations, fractalfract, fuels, future, futureinternet, futureparasites, futurepharmacol, futurephys, futuretransp, galaxies, games, gases, gastroent, gastrointestdisord, gastronomy, gels, genealogy, genes, geographies, geohazards, geomatics, geometry, geosciences, geotechnics, geriatrics, glacies, grasses, greenhealth, gucdd, hardware, hazardousmatters, healthcare, hearts, hemato, hematolrep, heritage, higheredu, highthroughput, histories, horticulturae, hospitals, humanities, humans, hydrobiology, hydrogen, hydrology, hygiene, idr, iic, ijerph, ijfs, ijgi, ijmd, ijms, ijns, ijpb, ijt, ijtm, ijtpp, ime, immuno, informatics, information, infrastructures, inorganics, insects, instruments, inventions, iot, j, jal, jcdd, jcm, jcp, jcs, jcto, jdad, jdb, jeta, jfb, jfmk, jimaging, jintelligence, jlpea, jmahp, jmmp, jmms, jmp, jmse, jne, jnt, jof, joitmc, joma, jop, jor, journalmedia, jox, jpbi, jpm, jrfm, jsan, jtaer, jvd, jzbg, kidney, kidneydial, kinasesphosphatases, knowledge, labmed, laboratories, land, languages, laws, life, lights, limnolrev, lipidology, liquids, literature, livers, logics, logistics, lubricants, lymphatics, machines, macromol, magnetism, magnetochemistry, make, marinedrugs, materials, materproc, mathematics, mca, measurements, medicina, medicines, medsci, membranes, merits, metabolites, metals, meteorology, methane, metrics, metrology, micro, microarrays, microbiolres, microelectronics, micromachines, microorganisms, microplastics, microwave, minerals, mining, mmphys, modelling, molbank, molecules, mps, msf, mti, multimedia, muscles, nanoenergyadv, nanomanufacturing, nanomaterials, ncrna, ndt, network, neuroglia, neurolint, neurosci, nitrogen, notspecified, nursrep, nutraceuticals, nutrients, obesities, oceans, ohbm, onco, oncopathology, optics, oral, organics, organoids, osteology, oxygen, parasites, parasitologia, particles, pathogens, pathophysiology, pediatrrep, pets, pharmaceuticals, pharmaceutics, pharmacoepidemiology, pharmacy, philosophies, photochem, photonics, phycology, physchem, physics, physiologia, plants, plasma, platforms, pollutants, polymers, polysaccharides, populations, poultry, powders, preprints, proceedings, processes, prosthesis, proteomes, psf, psych, psychiatryint, psychoactives, psycholint, publications, purification, quantumrep, quaternary, qubs, radiation, reactions, realestate, receptors, recycling, regeneration, religions, remotesensing, reports, reprodmed, resources, rheumato, risks, robotics, rsee, ruminants, safety, sci, scipharm, sclerosis, seeds, sensors, separations, sexes, signals, sinusitis, siuj, skins, smartcities, sna, societies, socsci, software, soilsystems, solar, solids, spectroscj, sports, standards, stats, std, stresses, surfaces, surgeries, suschem, sustainability, symmetry, synbio, systems, tae, targets, taxonomy, technologies, telecom, test, textiles, thalassrep, therapeutics, thermo, timespace, tomography, tourismhosp, toxics, toxins, transplantology, transportation, traumacare, traumas, tropicalmed, universe, urbansci, uro, vaccines, vehicles, venereology, vetsci, vibration, virtualworlds, viruses, vision, waste, water, wem, wevj, wild, wind, women, world, youth, zoonoticdis

%---------
% article
%---------
% The default type of manuscript is "article", but can be replaced by: 
% abstract, addendum, article, benchmark, book, bookreview, briefcommunication, briefreport, casereport, changes, clinicopathologicalchallenge, comment, commentary, communication, conceptpaper, conferenceproceedings, correction, conferencereport, creative, datadescriptor, discussion, entry, expressionofconcern, extendedabstract, editorial, essay, erratum, fieldguide, hypothesis, interestingimages, letter, meetingreport, monograph, newbookreceived, obituary, opinion, proceedingpaper, projectreport, reply, retraction, review, perspective, protocol, shortnote, studyprotocol, supfile, systematicreview, technicalnote, viewpoint, guidelines, registeredreport, tutorial,  giantsinurology, urologyaroundtheworld
% supfile = supplementary materials

%----------
% submit
%----------
% The class option "submit" will be changed to "accept" by the Editorial Office when the paper is accepted. This will only make changes to the frontpage (e.g., the logo of the journal will get visible), the headings, and the copyright information. Also, line numbering will be removed. Journal info and pagination for accepted papers will also be assigned by the Editorial Office.

%------------------
% moreauthors
%------------------
% If there is only one author the class option oneauthor should be used. Otherwise use the class option moreauthors.

%---------
% pdftex
%---------
% The option pdftex is for use with pdfLaTeX. Remove "pdftex" for (1) compiling with LaTeX & dvi2pdf (if eps figures are used) or for (2) compiling with XeLaTeX.

%=================================================================
% MDPI internal commands - do not modify
\firstpage{1} 
\makeatletter 
\setcounter{page}{\@firstpage} 
\makeatother
\pubvolume{1}
\issuenum{1}
\articlenumber{0}
\pubyear{2025}
\copyrightyear{2025}
%\externaleditor{Firstname Lastname} % More than 1 editor, please add `` and '' before the last editor name
\datereceived{ } 
\daterevised{ } % Comment out if no revised date
\dateaccepted{ } 
\datepublished{ } 
%\datecorrected{} % For corrected papers: "Corrected: XXX" date in the original paper.
%\dateretracted{} % For retracted papers: "Retracted: XXX" date in the original paper.
\hreflink{https://doi.org/} % If needed use \linebreak
%\doinum{}
%\pdfoutput=1 % Uncommented for upload to arXiv.org
%\CorrStatement{yes}  % For updates
%\longauthorlist{yes} % For many authors that exceed the left citation part
%\IsAssociation{yes} % For association journals

%=================================================================
% Add packages and commands here. The following packages are loaded in our class file: fontenc, inputenc, calc, indentfirst, fancyhdr, graphicx, epstopdf, lastpage, ifthen, float, amsmath, amssymb, lineno, setspace, enumitem, mathpazo, booktabs, titlesec, etoolbox, tabto, xcolor, colortbl, soul, multirow, microtype, tikz, totcount, changepage, attrib, upgreek, array, tabularx, pbox, ragged2e, tocloft, marginnote, marginfix, enotez, amsthm, natbib, hyperref, cleveref, scrextend, url, geometry, newfloat, caption, draftwatermark, seqsplit
% cleveref: load \crefname definitions after \begin{document}

%=================================================================
% Please use the following mathematics environments: Theorem, Lemma, Corollary, Proposition, Characterization, Property, Problem, Example, ExamplesandDefinitions, Hypothesis, Remark, Definition, Notation, Assumption
%% For proofs, please use the proof environment (the amsthm package is loaded by the MDPI class).

%=================================================================
% Full title of the paper (Capitalized)
\Title{Title}

% MDPI internal command: Title for citation in the left column
\TitleCitation{Title}

% Author Orchid ID: enter ID or remove command
\newcommand{\orcidauthorA}{0009-0007-3141-5160} % Add \orcidA{} behind the author's name
%\newcommand{\orcidauthorB}{0000-0000-0000-000X} % Add \orcidB{} behind the author's name

% Authors, for the paper (add full first names)
\Author{Allan Bolanos Barrientos $^{1}$\orcidA{}, Martín Solís $^{2}$ and Firstname Lastname $^{2,}$*}

%\longauthorlist{yes}

% MDPI internal command: Authors, for metadata in PDF
\AuthorNames{Firstname Lastname, Firstname Lastname and Firstname Lastname}

% Author citation:  
\AuthorCitation{Lastname, F.; Lastname, F.; Lastname, F.}

% Affiliations / Addresses (Add [1] after \address if there is only one affiliation.)
\address{%
$^{1}$ \quad Costa Rica Institute of Technology. School of Computer Engineering; a.bolanos.2@estudiantec.cr\\
$^{2}$ \quad Costa Rica Institute of Technology; marsolis@itcr.ac.cr}

% Contact information of the corresponding author
\corres{Correspondence: marsolis@itcr.ac.cr}

% Current address and/or shared authorship
%\firstnote{Current address: Affiliation.}  
% Current address should not be the same as any items in the Affiliation section.

%\secondnote{These authors contributed equally to this work.}
% The commands \thirdnote{} till \eighthnote{} are available for further notes.

%\simplesumm{} % Simple summary

%\conference{} % An extended version of a conference paper

% Abstract (Do not insert blank lines, i.e. \\) 
\abstract{A single paragraph of about 200 words maximum.}

% Keywords
\keyword{keyword 1; keyword 2; keyword 3 (List three to ten pertinent keywords specific to the article; yet reasonably common within the subject discipline.)} 

% The fields PACS, MSC, and JEL may be left empty or commented out if not applicable
%\PACS{J0101}
%\MSC{}
%\JEL{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Diversity
%\LSID{\url{http://}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Applied Sciences
%\featuredapplication{Authors are encouraged to provide a concise description of the specific application or a potential application of the work. This section is not mandatory.}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Data
%\dataset{DOI number or link to the deposited data set if the data set is published separately. If the data set shall be published as a supplement to this paper, this field will be filled by the journal editors. In this case, please submit the data set as a supplement.}
%\datasetlicense{License under which the data set is made available (CC0, CC-BY, CC-BY-SA, CC-BY-NC, etc.)}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal BioTech, Fishes, Neuroimaging and Toxins
%\keycontribution{The breakthroughs or highlights of the manuscript. Authors can write one or two sentences to describe the most important part of the paper.}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Encyclopedia
%\encyclopediadef{For entry manuscripts only: please provide a brief overview of the entry title instead of an abstract.}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Different journals have different requirements. Please check the specific journal guidelines in the "Instructions for Authors" on the journal's official website.
%\addhighlights{yes}
%\renewcommand{\addhighlights}{%
%
%\noindent The goal is to increase the discoverability and readability of the article via search engines and other scholars. Highlights should not be a copy of the abstract, but a simple text allowing the reader to quickly and simplified find out what the article is about and what can be cited from it. Each of these parts should be devoted up to 2~bullet points.\vspace{3pt}\\
%\textbf{What are the main findings?}
% \begin{itemize}[labelsep=2.5mm,topsep=-3pt]
% \item First bullet.
% \item Second bullet.
% \end{itemize}\vspace{3pt}
%\textbf{What is the implication of the main finding?}
% \begin{itemize}[labelsep=2.5mm,topsep=-3pt]
% \item First bullet.
% \item Second bullet.
% \end{itemize}
%}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\setcounter{section}{0} %% Remove this when starting to work on the template.


\section{Introduction}

Change point detection is the finding of changes in the behavior pattern in a time series. The interest in finding change points can be online or offline. Online algorithms process data within a sliding window with size n to detect change points in real, while offline the entire time series is processed at once to detect the change points in the complete time series \cite{aminikhanghahi2017survey}. Online change detection has increased its relevance due to the rapid growth of stream data generation in diverse domains \cite{namoano2019online}. Nowadays the development of algorithms in online detection has contributed to relevant important issues such as fuel leakage detection \cite{chu2025real}, detection of change points in the spread of viruses to reveal the effectiveness of interventions \cite{dehning2020inferring}, detect pipe burst localization in water distribution \cite{mzembegwa2024real}, etc. 
Change point detection methods could perform poorly when the data are noisy \cite{chen2016general} and make detection points challenging \cite{gold2018doubly}. This problem becomes greater when the changing points are subtle \cite{chu2025real}, even for humans. For example, in figure xx it can be seen the difficulty to the change point detection for the same time series with different levels of noise. Besides, the presence of noise can lead to the detection of false change points, when what there is noise. 
Although noise can influence the performance of change detection algorithms, there is no benchmarking of how the algorithms perform in the presence of different levels noise levels in contexts with weak and strong change points, as far as we know. Therefore, the present research analyzes how the more common online change detection algorithms perform in the presence of noise and different levels of change.  The findings obtained can lead to a better choice of detection methods according to the type of time series under study, as well as opening new research questions on how to improve the detection of change points.
As a second objective, this research will address a real problem such as the task of change points detection in time series of crimes occurrences. Although this task can contribute to monitoring crime, detecting regions where the pattern of crime changes, and establishing police strategies, it has received little attention on the research \cite{konstantinou2023trend}. On the other hand, the time series of crimes tend to show relevant levels of noise which can make it difficult to detect change points. Crime time series depend on the citizen's willingness to report the crime. Thus, noise does not only come from randomness of the phenomena but from a structural bias in the collection of information.

In summary the novel contributions of the manuscript are the next:
\begin{itemize}
    \item A novel performance benchmarking of the online change point detection algorithms based on the level of noise and strength of the change point.
    \item Analysis of the change point detection in crime time series according to the level of noise.
    \item A new labeled dataset to detect change points in crime time series .
\end{itemize}


\section{Literature review}
Recent studies have addressed the issue of online change point detection.  In \cite{cakmak2024benchmarking} six commonly available state-of-the-art changepoint detection approaches, with two additional modified algorithms, were compared in cardiovascular time series data. In \cite{van2020evaluation} a benchmarking of popular algorithms was generated using simulated data and 37 real time series from various application domains. \cite{wang2021online} evaluated online changepoint detection algorithms that work on unbounded data stream with a constant time and space complexity. Other authors, instead of comparing known methods in different contexts, developed a new proposal for online change point detection. These are the cases of \cite{chu2025real}, \cite{zameni2020unsupervised}, \cite{gold2018doubly}. In \cite{chu2025real}, the authors proposed a novel memory-based online change point detection (MOCPD) framework to find fuel leakage detection delays. \cite{gold2018doubly} create a method called Delta point that divides the time series into intervals of user-specified, domain specific length for which a suspected change point may be contained. In \cite{zameni2020unsupervised} the method created does not require any prior distributional knowledge of the time series and exploits the Information Gain to verify each new candidate change point score. In the studies where a benchmarking of methods is generated, neither in the studies where new proposals are created are there evaluations of the algorithm’s performance according to the level of noise and the level of change point. Therefore, there is a lack of knowledge about how algorithms work in specific circumstances.
In relation to the detection of changes in crime events patterns, only a few studies have analyzed its effectiveness, although the application of change point detection algorithms can contribute to the police monitoring of criminality. This is the case of \cite{konstantinou2023trend} who investigate the effectiveness of both online and offline change point detection methods towards identifying critical changes in crime-related time series from Boston’ and the ‘London Police Records’ datasets. The best performance is obtained with BOCPD. In \cite{albertetti2016change} the authors proposed a fuzzy approach to detect change points in the CICOP data set (crime real-world data), that consisting of 32 monthly time series. They conclude that the proposal method has great potential in crime analysis, however there are not comparisons with baselines or other methods.  The work of \cite{theodosiadou2021change} consists in a proposal towards detecting change points in terrorism-related time series. They conclude that the proposed framework could be seen as an alternative way to identify links between terrorism and online activity


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Materials and Methods}

This section describes the comprehensive methodology employed to benchmark online change point detection algorithms. The experimental design comprises three complementary evaluation approaches: (1) controlled experiments using synthetic time series with known change points, (2) real-world evaluation using manually labeled crime occurrence data, and (3) benchmark evaluation on the TCPD (Time Series Change Point Database) repository.

\subsection{Implementation Framework}

All experiments were implemented in Python 3.13 using a modular pipeline architecture. The core dependencies include NumPy (1.26+) for numerical computation, Pandas (2.0+) for data manipulation, and SciPy (1.11+) for signal processing. The complete source code is organized in a structured repository with separate modules for data generation, algorithm implementations, evaluation metrics, and benchmarking pipelines.

\subsection{Change Point Detection Algorithms}

A comprehensive suite of 17 online change point detection algorithms was evaluated, representing diverse methodological paradigms. Table~\ref{tab:algorithms} summarizes the evaluated algorithms, organized by approach and implementation library.

\begin{table}[H]
\caption{Summary of 17 evaluated change point detection algorithms.\label{tab:algorithms}}
\centering
\begin{tabular}{llll}
\toprule
\textbf{Algorithm} & \textbf{Method} & \textbf{Library} & \textbf{Approach} \\
\midrule
ADWIN & Adaptive Windowing & River & Statistical \\
Page-Hinkley & Sequential likelihood ratio & River & Statistical \\
EWMA-NumPy & Exponential smoothing & NumPy & Statistical \\
EWMA-OCPDet & Adaptive EWMA & OCPDet & Statistical \\
CUSUM & Cumulative sum control & OCPDet & Statistical \\
\midrule
SSM-Canary & State-space model & Canary & Model-based \\
SKF-Canary & Square root Kalman & Canary & Model-based \\
TAGI-LSTM-SSM & Neural-augmented SSM & Canary & Model-based \\
\midrule
BOCPD & Bayesian online CPD & CPFinder & Bayesian \\
\midrule
ChangeFinder & AR-based outlier scoring & SDAR & Density \\
RULSIF & Density ratio estimation & Roerich & Density \\
\midrule
Focus & PELT with RBF cost & Changepoint & Segmentation \\
Gaussian & PELT with L2 cost & Changepoint & Segmentation \\
NPFocus & Non-parametric kernel & Changepoint & Segmentation \\
MDFocus & Multivariate RBF & Changepoint & Segmentation \\
\midrule
Neural-Net & MLP-based detector & OCPDet & ML \\
Two-Sample Tests & Hypothesis testing & OCPDet & ML \\
\bottomrule
\end{tabular}
\end{table}

\subsubsection{Statistical Control Charts}

\textbf{ADWIN (Adaptive Windowing)} from the River library implements adaptive window management for concept drift detection. Key hyperparameters include delta ($\delta \in \{0.002, 0.005, 0.01\}$) controlling sensitivity, clock period ($\{32, 64\}$), and grace period ($\{10, 20\}$) for initialization.

\textbf{Page-Hinkley Test} (River) detects changes in sequential data using cumulative sums. Hyperparameters include threshold ($\{20, 40, 50\}$), minimum instances ($\{5, 10, 20\}$), and drift tolerance delta ($\{0.001, 0.005\}$).

\textbf{EWMA (Exponentially Weighted Moving Average)} was evaluated in two implementations:
\begin{itemize}
    \item \textbf{EWMA-NumPy}: Lightweight NumPy-based implementation with smoothing parameter alpha ($\alpha \in \{0.05, 0.1, 0.2\}$), detection threshold ($\{2.0, 2.5\}$), and minimum instances ($\{5, 10\}$).
    \item \textbf{EWMA-OCPDet}: Adaptive version from OCPDet with alpha ($\{0.05, 0.1, 0.2\}$), threshold ($\{2.0, 3.0\}$), and minimum instances ($\{5\}$).
\end{itemize}

\textbf{CUSUM (Cumulative Sum)} from OCPDet library uses sequential likelihood ratio testing with threshold ($\{6.0, 10.0\}$), drift parameter ($\{0.0\}$), and minimum distance between detections ($\{20, 30\}$).

\subsubsection{Model-Based Detection}

\textbf{State-Space Models (SSM)} using the Canary library model time series as latent state evolution with observation noise. Three variants were evaluated:
\begin{itemize}
    \item \textbf{SSM-Canary}: Basic SSM with process noise ($\{1 \times 10^{-3}, 5 \times 10^{-3}\}$), measurement noise ($\{0.4, 0.6, 0.8\}$), threshold ($\{2.5, 3.0\}$).
    \item \textbf{SKF (Square Root Kalman Filter)}: Numerically stable variant with process noise ($\{5 \times 10^{-3}, 1 \times 10^{-2}\}$), measurement noise ($\{0.6, 0.9\}$), threshold ($\{3.0, 3.5\}$).
    \item \textbf{TAGI-LSTM-SSM}: Neural network-augmented SSM combining Tractable Approximate Gaussian Inference with LSTM architecture. Hyperparameters include process/measurement noise, threshold, and adaptation rate ($5 \times 10^{-5}$).
\end{itemize}

\subsubsection{Bayesian Methods}

\textbf{Bayesian Online Change Point Detection (BOCPD)} via CPFinder implements the Adams-MacKay algorithm using Student's t-distribution for likelihood. Hyperparameters include hazard function parameter ($\lambda \in \{150, 300\}$), prior parameters alpha ($\{0.1, 0.3\}$), beta ($0.01$), kappa ($1.0$), probability threshold ($\{0.5, 0.7\}$), and minimum distance ($25$).

\subsubsection{Density Estimation Methods}

\textbf{ChangeFinder} (SDAR implementation) uses auto-regressive modeling and outlier scoring. Hyperparameters include learning rate r ($\{0.3, 0.5, 0.7\}$), AR order ($\{1, 2\}$), smoothing window ($\{5, 7, 9\}$), threshold ($\{2.0, 2.5\}$), and minimum distance ($\{20, 25\}$).

\textbf{RULSIF (Relative Unconstrained Least-Squares Importance Fitting)} from Roerich library estimates density ratios between reference and test windows. Hyperparameters include window size ($\{5, 10\}$), lag size ($\{60, 90\}$), step ($\{2, 3\}$), epochs ($\{1, 2\}$), threshold ($\{0.08, 0.10, 0.12\}$), minimum distance ($\{25, 30\}$), and alpha ($\{0.05, 0.1\}$).

\subsubsection{Segmentation-Based Methods}

\textbf{Changepoint-Online} library provides multiple cost functions with PELT (Pruned Exact Linear Time) algorithm:
\begin{itemize}
    \item \textbf{Focus}: RBF kernel cost with penalty ($\{20, 30\}$), min\_size ($\{15, 20, 25\}$), jump ($\{3, 5\}$).
    \item \textbf{Gaussian}: L2 cost for Gaussian distributions with penalty ($\{20, 30, 40\}$), min\_size ($\{15, 20\}$), jump ($\{3, 5\}$).
    \item \textbf{NPFocus}: Non-parametric kernel-based with penalty ($\{20, 30\}$), width ($\{30, 40, 50\}$), jump ($\{3\}$).
    \item \textbf{MDFocus}: Multivariate detection with penalty ($\{20, 30\}$), min\_size ($\{20, 30\}$), jump ($\{3\}$).
\end{itemize}

\subsubsection{Machine Learning Approaches}

\textbf{Neural Network Detector} (OCPDet) uses multilayer perceptrons to learn change patterns. Hyperparameters include window size ($\{20, 30\}$), step ($\{1, 3\}$), hidden layer architecture ($\{(20,), (30, 15), (25, 12)\}$), threshold ($\{2.0, 2.5\}$), and minimum distance ($\{25, 30\}$).

\textbf{Two-Sample Tests} (OCPDet) applies statistical hypothesis testing on sliding windows using Kolmogorov-Smirnov or Mann-Whitney U tests. Parameters include window size ($\{30, 40, 50\}$), step ($\{5, 10\}$), significance level alpha ($\{0.01, 0.05\}$), and minimum distance ($\{20, 30\}$).

\subsection{Benchmark 1: Synthetic Data Evaluation}

\subsubsection{Data Generation Process}

Synthetic time series were generated using a parametric framework to systematically control change characteristics. The generation process follows:

\textbf{Series Structure:} Each series has length $n \in \{200, 300, 400\}$ observations with $k \in \{1, 2, 3\}$ change points positioned randomly with minimum separation of $\lfloor n/4 \rfloor$ to ensure distinct segments.

\textbf{Change Type Generation:} Two fundamental change types were implemented:
\begin{linenomath}
\begin{equation}
y_t^{escalon} = \begin{cases}
\mu_i & \text{if } t_{c_{i-1}} \leq t < t_{c_i}
\end{cases}
\end{equation}
\end{linenomath}

\begin{linenomath}
\begin{equation}
y_t^{pendiente} = \mu_i + \frac{(\mu_{i+1} - \mu_i)}{w} \cdot (t - t_{c_i}) \quad \text{for } t_{c_i} \leq t < t_{c_i} + w
\end{equation}
\end{linenomath}

where $w = \min(20, \lfloor (t_{c_{i+1}} - t_{c_i})/2 \rfloor)$ is the transition window length.

\textbf{Noise and Change Magnitude Control:} Additive Gaussian noise $\epsilon_t \sim \mathcal{N}(0, \sigma^2)$ was added to create controlled conditions:
\begin{itemize}
    \item \textbf{High Noise}: NSR $\in [3.0, 6.0]$ (noisy conditions)
    \item \textbf{Low Noise}: NSR $\in [0.0, 0.4]$ (clean conditions)
    \item \textbf{High Change}: Magnitude $\Delta \mu \in [3.0, 6.0]$ (strong changes)
    \item \textbf{Low Change}: Magnitude $\Delta \mu \in [0.5, 1.5]$ (subtle changes)
\end{itemize}

\textbf{Dataset Composition:} The complete synthetic dataset consists of:
\begin{itemize}
    \item 2 change types (escalón, pendiente)
    \item 2 noise levels (alto, bajo)
    \item 2 change strengths (alto, bajo)
    \item 15 replications per combination
    \item Total: $2 \times 2 \times 2 \times 15 = 120$ series
\end{itemize}

\subsubsection{Train-Test Split}

Series were randomly divided (50\%-50\%) into training and test sets, stratified by change type, noise level, and change strength to ensure balanced representation. Random seed was fixed (seed=123) for reproducibility.

\subsubsection{Hyperparameter Optimization Protocol}

For each algorithm, exhaustive grid search was performed on the training set:
\begin{enumerate}
    \item \textbf{Grid Definition}: Algorithm-specific grids were defined based on literature recommendations and preliminary experiments (see Table~\ref{tab:param\_grids}).
    \item \textbf{Training Evaluation}: Each parameter combination was applied to all training series, computing F1-score with tolerance $\tau = 10$ time steps.
    \item \textbf{Best Configuration}: The configuration maximizing mean F1-score across training series was selected.
    \item \textbf{Test Application}: Selected configuration was applied to test set for final evaluation.
\end{enumerate}

\begin{table}[H]
\caption{Example hyperparameter grids for selected algorithms.\label{tab:param\_grids}}
\centering
\begin{tabular}{lll}
\toprule
\textbf{Algorithm} & \textbf{Hyperparameter} & \textbf{Values} \\
\midrule
ADWIN & delta & \{0.002, 0.005, 0.01\} \\
 & clock & \{32, 64\} \\
 & grace\_period & \{10, 20\} \\
\midrule
Page-Hinkley & threshold & \{20, 40, 50\} \\
 & min\_instances & \{5, 10, 20\} \\
 & delta & \{0.001, 0.005\} \\
\midrule
EWMA & alpha & \{0.05, 0.1, 0.2\} \\
 & threshold & \{2.0, 2.5, 3.0\} \\
 & min\_instances & \{5, 10\} \\
\midrule
BOCPD & hazard\_lambda & \{150, 300\} \\
 & alpha & \{0.1, 0.3\} \\
 & prob\_threshold & \{0.5, 0.7\} \\
\midrule
ChangeFinder & r (learning rate) & \{0.3, 0.5, 0.7\} \\
 & order & \{1, 2\} \\
 & smooth & \{5, 7, 9\} \\
\midrule
SSM-Canary & process\_noise & \{1e-3, 5e-3\} \\
 & measurement\_noise & \{0.4, 0.6, 0.8\} \\
 & threshold & \{2.5, 3.0\} \\
\midrule
Focus & penalty & \{20, 30, 40\} \\
 & min\_size & \{15, 20, 25\} \\
 & jump & \{3, 5\} \\
\bottomrule
\end{tabular}
\end{table}

\subsection{Benchmark 2: Real Crime Data Evaluation}

\subsubsection{Data Collection and Annotation}

Real-world time series were constructed from crime occurrence records from Costa Rican criminal databases, covering January 2015 to December 2023. The annotation process involved:

\textbf{Series Construction:}
\begin{itemize}
    \item \textbf{Aggregation}: Events aggregated into weekly counts
    \item \textbf{Selection}: 50 series selected representing diverse crime types and geographic regions
    \item \textbf{Length}: Series range from 60 to 400 observations (median: 120)
    \item \textbf{Normalization}: Standardized to zero mean and unit variance
\end{itemize}

\textbf{Manual Annotation Protocol:}
Two expert annotators (authors Allan and Martín) independently labeled change points following these criteria:
\begin{itemize}
    \item \textbf{Visual Inspection}: Identifying significant shifts in mean, variance, or trend
    \item \textbf{Change Type Classification}: Labeling as step (escalón) or slope (pendiente)
    \item \textbf{Consensus Resolution}: Disagreements resolved through discussion
    \item \textbf{Metadata Recording}: Ground truth stored in CSV format with columns: filename, series\_id, annotator, changepoint\_locations, changepoint\_types
\end{itemize}

\subsubsection{Automatic Classification System}

To enable systematic performance analysis, all crime series were automatically classified using three dimensions:

\textbf{Noise Level Estimation via NSR:}
Signal smoothing using Savitzky-Golay filter (polynomial order 2, window $w = \max(5, \lfloor 0.05n \rfloor)$):
\begin{linenomath}
\begin{equation}
NSR = \frac{\text{std}(x - \hat{y})}{\text{std}(\hat{y})}
\end{equation}
\end{linenomath}

Classification: High noise if $NSR > \text{median}(NSR)$, otherwise low noise.

\textbf{Change Magnitude Estimation:}
For segments defined by change points $\{t_{c_1}, \ldots, t_{c_k}\}$:
\begin{linenomath}
\begin{equation}
\Delta_{\max} = \max_{i} |\bar{x}_{i+1} - \bar{x}_i|
\end{equation}
\end{linenomath}

Classification: High change if $\Delta_{\max} > \text{median}(\Delta_{\max})$, otherwise low change.

\textbf{Change Type Extraction:} Extracted from manual annotations (escalón/pendiente).

\textbf{Combined Classification:} This yields 8 categories: 2 change types $\times$ 2 noise levels $\times$ 2 change magnitudes.

\subsubsection{Evaluation Protocol}

Same train-test split and grid search methodology as synthetic benchmark, adapted for the 50 labeled series (25 training, 25 test).

\subsection{Benchmark 3: TCPD Repository Evaluation}

\subsubsection{Dataset Description}

The Time Series Change Point Database (TCPD) repository contains diverse real-world time series from various domains without ground truth annotations. We evaluated algorithms on all available TCPD datasets located in \texttt{data/TCDP-paper/} directory.

\subsubsection{Evaluation Methodology}

Since TCPD series lack ground truth labels, evaluation focused on:
\begin{itemize}
    \item \textbf{Detection Count}: Number of change points detected per series
    \item \textbf{Detection Consistency}: Variability across different parameter settings
    \item \textbf{Runtime Performance}: Execution time per series (timeout: 120 seconds)
    \item \textbf{Algorithm Robustness}: Success rate (non-crash, non-timeout)
\end{itemize}

Simplified parameter grids (fewer values per hyperparameter) were used due to lack of ground truth for optimization.

\subsection{Performance Metrics}

Three complementary metrics were computed for benchmarks with ground truth (synthetic and crime data):

\textbf{F1-Score with Tolerance:} For tolerance window $\tau$:
\begin{linenomath}
\begin{equation}
\text{Precision} = \frac{|\{d \in D : \exists t \in T, |d-t| \leq \tau\}|}{|D|}
\end{equation}
\end{linenomath}

\begin{linenomath}
\begin{equation}
\text{Recall} = \frac{|\{t \in T : \exists d \in D, |t-d| \leq \tau\}|}{|T|}
\end{equation}
\end{linenomath}

\begin{linenomath}
\begin{equation}
F1 = 2 \cdot \frac{\text{Precision} \cdot \text{Recall}}{\text{Precision} + \text{Recall}}
\end{equation}
\end{linenomath}

where $D$ is the set of detected change points, $T$ is the set of true change points, and $\tau = 10$ time steps.

\textbf{Mean Time to Detection (MTTD):}
\begin{linenomath}
\begin{equation}
MTTD = \frac{1}{|TP|} \sum_{i \in TP} (t_{\text{detected}_i} - t_{\text{true}_i})
\end{equation}
\end{linenomath}

where $TP$ is the set of true positives (correctly detected change points within tolerance).

\textbf{Maximum Mean Discrepancy (MMD):} Average distance between true and detected change point sets, penalizing both missed detections and false positives.

\subsection{Experimental Infrastructure}

\textbf{Hardware:} All experiments executed on [INSERT SPECIFICATIONS].

\textbf{Software:} Python 3.13, NumPy 1.26.3, Pandas 2.1.4, SciPy 1.11.4, River 0.21.0, Ruptures 1.1.9, OCPDet 1.0.0, Canary 0.3.0, CPFinder 0.2.0, Roerich 1.0.0.

\textbf{Reproducibility:} Random seeds fixed, configuration files versioned, results timestamped with format \texttt{MM-DD-YYYY-resultados\_*.csv}.

\textbf{Runtime Management:} Timeout mechanisms (120s per algorithm-series pair) prevent infinite loops; failed runs recorded with error codes.

\textbf{Output Format:} Results saved in structured CSV (tabular metrics) and JSON (detailed per-series results with metadata) for downstream analysis.

\subsection{Comparative Analysis Framework}

Results were analyzed across multiple dimensions to provide comprehensive insights:

\textbf{Overall Performance Ranking:} Algorithms ranked by mean F1-score across all test series within each benchmark.

\textbf{Category-Specific Analysis:} Performance stratified by:
\begin{itemize}
    \item Change type (escalón vs. pendiente)
    \item Noise level (high vs. low)
    \item Change magnitude (high vs. low)
    \item Combined categories (8 groups)
\end{itemize}

\textbf{Best-in-Class Identification:} For each category, the algorithm with highest F1-score was identified, along with the corresponding series exhibiting best detection quality.

\textbf{Robustness Assessment:} Standard deviation of F1-scores across series within each category, indicating algorithm stability.

\textbf{Detection Speed Analysis:} MTTD compared across algorithms to identify fastest detectors.

\textbf{Library Comparison:} Performance aggregated by implementation library to assess framework quality.

The complete experimental workflow is summarized in Figure~\ref{fig:workflow} (conceptual), showing data flow from generation/collection through classification, optimization, evaluation, and analysis phases.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Results}

This section may be divided by subheadings. It should provide a concise and precise description of the experimental results, their interpretation as well as the experimental conclusions that can be drawn.
\subsection{Subsection}
\subsubsection{Subsubsection}

Bulleted lists look like this:
\begin{itemize}
\item	First bullet;
\item	Second bullet;
\item	Third bullet.
\end{itemize}



%\begin{listing}[H]
%\caption{Title of the listing}
%\rule{\columnwidth}{1pt}
%\raggedright Text of the listing. In font size footnotesize, small, or normalsize. Preferred format: left aligned and single spaced. Preferred border format: top border line and bottom border line.
%\rule{\columnwidth}{1pt}
%\end{listing}

Text.

Text.

\subsection{Formatting of Mathematical Components}

This is the example 1 of equation:
\begin{linenomath}
\begin{equation}
a = 1,
\end{equation}
\end{linenomath}

%% If the documentclass option "submit" is chosen, please insert a blank line before and after any math environment (equation and eqnarray environments). This ensures correct linenumbering. The blank line should be removed when the documentclass option is changed to "accept" because the text following an equation should not be a new paragraph.


%\isPreprints{}{% This command is only used for ``preprints''.

%} % If the paper is ``preprints'', please uncomment this parenthesis.

%% Example of a page in landscape format (with table and table footnote).
%\startlandscape
%\begin{table}[H] %% Table in wide page
%%\isPreprints{\centering}{} % This command is only used for ``preprints''.
%\caption{This is a very wide table.\label{tab3}}
%	\begin{tabularx}{\textwidth}{CCCC}
%		\toprule
%		\textbf{Title 1}	& \textbf{Title 2}	& \textbf{Title 3}	& \textbf{Title 4}\\
%		\midrule
%		Entry 1		& Data			& Data			& This cell has some longer content that runs over two lines.\\
%		Entry 2		& Data			& Data			& Data\textsuperscript{1}\\
%		\bottomrule
%	\end{tabularx}
%%\isPreprints{}{% This command is only used for ``preprints''.
%	\begin{adjustwidth}{+\extralength}{0cm}
%%} % If the paper is ``preprints'', please uncomment this parenthesis.
%		\noindent\footnotesize{\textsuperscript{1} This is a table footnote.}
%%\isPreprints{}{% This command is only used for ``preprints''.
%	\end{adjustwidth}
%%} % If the paper is ``preprints'', please uncomment this parenthesis.
%\end{table}
%\finishlandscape


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Discussion}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Conclusions}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Patents}

This section is not mandatory, but may be added if there are patents resulting from the work reported in this manuscript.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\vspace{6pt} 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% optional
%\supplementary{The following supporting information can be downloaded at:  \linksupplementary{s1}, Figure S1: title; Table S1: title; Video S1: title.}

% Only for journal Methods and Protocols:
% If you wish to submit a video article, please do so with any other supplementary material.
% \supplementary{The following supporting information can be downloaded at: \linksupplementary{s1}, Figure S1: title; Table S1: title; Video S1: title. A supporting video article is available at doi: link.}

% Only used for preprtints:
% \supplementary{The following supporting information can be downloaded at the website of this paper posted on \href{https://www.preprints.org/}{Preprints.org}.}

% Only for journal Hardware:
% If you wish to submit a video article, please do so with any other supplementary material.
% \supplementary{The following supporting information can be downloaded at: \linksupplementary{s1}, Figure S1: title; Table S1: title; Video S1: title.\vspace{6pt}\\
%\begin{tabularx}{\textwidth}{lll}
%\toprule
%\textbf{Name} & \textbf{Type} & \textbf{Description} \\
%\midrule
%S1 & Python script (.py) & Script of python source code used in XX \\
%S2 & Text (.txt) & Script of modelling code used to make Figure X \\
%S3 & Text (.txt) & Raw data from experiment X \\
%S4 & Video (.mp4) & Video demonstrating the hardware in use \\
%... & ... & ... \\
%\bottomrule
%\end{tabularx}
%}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\authorcontributions{For research articles with several authors, a short paragraph specifying their individual contributions must be provided. The following statements should be used ``Conceptualization, X.X. and Y.Y.; methodology, X.X.; software, X.X.; validation, X.X., Y.Y. and Z.Z.; formal analysis, X.X.; investigation, X.X.; resources, X.X.; data curation, X.X.; writing---original draft preparation, X.X.; writing---review and editing, X.X.; visualization, X.X.; supervision, X.X.; project administration, X.X.; funding acquisition, Y.Y. All authors have read and agreed to the published version of the manuscript.'', please turn to the  \href{http://img.mdpi.org/data/contributor-role-instruction.pdf}{CRediT taxonomy} for the term explanation. Authorship must be limited to those who have contributed substantially to the work~reported.}

\funding{Please add: ``This research received no external funding'' or ``This research was funded by NAME OF FUNDER grant number XXX.'' and  and ``The APC was funded by XXX''. Check carefully that the details given are accurate and use the standard spelling of funding agency names at \url{https://search.crossref.org/funding}, any errors may affect your future funding.}

\institutionalreview{In this section, you should add the Institutional Review Board Statement and approval number, if relevant to your study. You might choose to exclude this statement if the study did not require ethical approval. Please note that the Editorial Office might ask you for further information. Please add “The study was conducted in accordance with the Declaration of Helsinki, and approved by the Institutional Review Board (or Ethics Committee) of NAME OF INSTITUTE (protocol code XXX and date of approval).” for studies involving humans. OR “The animal study protocol was approved by the Institutional Review Board (or Ethics Committee) of NAME OF INSTITUTE (protocol code XXX and date of approval).” for studies involving animals. OR “Ethical review and approval were waived for this study due to REASON (please provide a detailed justification).” OR “Not applicable” for studies not involving humans or animals.}

\informedconsent{Any research article describing a study involving humans should contain this statement. Please add ``Informed consent was obtained from all subjects involved in the study.'' OR ``Patient consent was waived due to REASON (please provide a detailed justification).'' OR ``Not applicable'' for studies not involving humans. You might also choose to exclude this statement if the study did not involve humans.

Written informed consent for publication must be obtained from participating patients who can be identified (including by the patients themselves). Please state ``Written informed consent has been obtained from the patient(s) to publish this paper'' if applicable.}

\dataavailability{We encourage all authors of articles published in MDPI journals to share their research data. In this section, please provide details regarding where data supporting reported results can be found, including links to publicly archived datasets analyzed or generated during the study. Where no new data were created, or where data is unavailable due to privacy or ethical restrictions, a statement is still required. Suggested Data Availability Statements are available in section ``MDPI Research Data Policies'' at \url{https://www.mdpi.com/ethics}.} 

% Only for journal Drones
%\durcstatement{Current research is limited to the [please insert a specific academic field, e.g., XXX], which is beneficial [share benefits and/or primary use] and does not pose a threat to public health or national security. Authors acknowledge the dual-use potential of the research involving xxx and confirm that all necessary precautions have been taken to prevent potential misuse. As an ethical responsibility, authors strictly adhere to relevant national and international laws about DURC. Authors advocate for responsible deployment, ethical considerations, regulatory compliance, and transparent reporting to mitigate misuse risks and foster beneficial outcomes.}

% Only for journal Nursing Reports
%\publicinvolvement{Please describe how the public (patients, consumers, carers) were involved in the research. Consider reporting against the GRIPP2 (Guidance for Reporting Involvement of Patients and the Public) checklist. If the public were not involved in any aspect of the research add: ``No public involvement in any aspect of this research''.}
%
%% Only for journal Nursing Reports
%\guidelinesstandards{Please add a statement indicating which reporting guideline was used when drafting the report. For example, ``This manuscript was drafted against the XXX (the full name of reporting guidelines and citation) for XXX (type of research) research''. A complete list of reporting guidelines can be accessed via the equator network: \url{https://www.equator-network.org/}.}
%
%% Only for journal Nursing Reports
%\useofartificialintelligence{Please describe in detail any and all uses of artificial intelligence (AI) or AI-assisted tools used in the preparation of the manuscript. This may include, but is not limited to, language translation, language editing and grammar, or generating text. Alternatively, please state that “AI or AI-assisted tools were not used in drafting any aspect of this manuscript”.}

\acknowledgments{In this section you can acknowledge any support given which is not covered by the author contribution or funding sections. This may include administrative and technical support, or donations in kind (e.g., materials used for experiments). Where GenAI has been used for purposes such as generating text, data, or graphics, or for study design, data collection, analysis, or interpretation of data, please add “During the preparation of this manuscript/study, the author(s) used [tool name, version information] for the purposes of [description of use]. The authors have reviewed and edited the output and take full responsibility for the content of this publication.”}

\conflictsofinterest{Declare conflicts of interest or state ``The authors declare no conflicts of interest.'' Authors must identify and declare any personal circumstances or interest that may be perceived as inappropriately influencing the representation or interpretation of reported research results. Any role of the funders in the design of the study; in the collection, analyses or interpretation of data; in the writing of the manuscript; or in the decision to publish the results must be declared in this section. If there is no role, please state ``The funders had no role in the design of the study; in the collection, analyses, or interpretation of data; in the writing of the manuscript; or in the decision to publish the results''.} 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional

%% Only for journal Encyclopedia
%\entrylink{The Link to this entry published on the encyclopedia platform.}

\abbreviations{Abbreviations}{
The following abbreviations are used in this manuscript:
\\

\noindent 
\begin{tabular}{@{}ll}
CPD & Change Point Detection\\
ADWIN & Adaptive Windowing\\
BOCPD & Bayesian Online Change Point Detection\\
CUSUM & Cumulative Sum\\
EWMA & Exponentially Weighted Moving Average\\
PELT & Pruned Exact Linear Time\\
SSM & State-Space Model\\
SKF & Square Root Kalman Filter\\
TAGI & Tractable Approximate Gaussian Inference\\
LSTM & Long Short-Term Memory\\
RULSIF & Relative Unconstrained Least-Squares Importance Fitting\\
NSR & Noise-to-Signal Ratio\\
MTTD & Mean Time to Detection\\
MMD & Maximum Mean Discrepancy\\
TCPD & Time Series Change Point Database\\
AR & Autoregressive\\
MLP & Multilayer Perceptron\\
RBF & Radial Basis Function
\end{tabular}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional
\appendixtitles{no} % Leave argument "no" if all appendix headings stay EMPTY (then no dot is printed after "Appendix A"). If the appendix sections contain a heading then change the argument to "yes".
\appendixstart
\appendix
\section[\appendixname~\thesection]{}
\subsection[\appendixname~\thesubsection]{}
The appendix is an optional section that can contain details and data supplemental to the main text---for example, explanations of experimental details that would disrupt the flow of the main text but nonetheless remain crucial to understanding and reproducing the research shown; figures of replicates for experiments of which representative data are shown in the main text can be added here if brief, or as Supplementary Data. Mathematical proofs of results not central to the paper can be added as an appendix.

\begin{table}[H] 
\caption{This is a table caption.\label{tab5}}
\centering
\begin{tabular}{ccc}
\toprule
\textbf{Title 1}	& \textbf{Title 2}	& \textbf{Title 3}\\
\midrule
Entry 1		& Data			& Data\\
Entry 2		& Data			& Data\\
\bottomrule
\end{tabular}
\end{table}

\section[\appendixname~\thesection]{}
All appendix sections must be cited in the main text. In the appendices, Figures, Tables, etc. should be labeled, starting with ``A''---e.g., Figure A1, Figure A2, etc.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\isPreprints{}{% This command is only used for ``preprints''.
\begin{adjustwidth}{-\extralength}{0cm}
%} % If the paper is ``preprints'', please uncomment this parenthesis.
%\printendnotes[custom] % Un-comment to print a list of endnotes

\reftitle{References}

% Please provide the correct journal abbreviation (e.g. according to the “List of Title Word Abbreviations” http://www.issn.org/services/online-services/access-to-the-ltwa/).
% Citations and References in Supplementary files are permitted provided that they also appear in the reference list here. 

%=====================================
% References, variant A: external bibliography
%=====================================
% \bibliography{your_external_BibTeX_file}

%=====================================
% References, variant B: internal bibliography
%=====================================

% ACS format
\begin{thebibliography}{999}

\bibitem{aminikhanghahi2017survey}
S. Aminikhanghahi and D. J. Cook,
``A survey of methods for time series change point detection,''
\textit{Knowledge and Information Systems}, vol.~51, no.~2, pp.~339--367, 2017.

\bibitem{namoano2019online}
B. Namoano, A. Starr, C. Emmanouilidis, and C. R. Carcel, ``Online change detection techniques in time series: An overview,'' in \emph{2019 IEEE International Conference on Prognostics and Health Management (ICPHM)}, 2019, pp. 1--10.

\bibitem{chu2025real}
R. Chu, L. Chik, Y. Song, J. Chan, and X. Li, ``Real-time fuel leakage detection via online change point detection,'' \emph{International Journal of Data Science and Analytics}, pp. 1--18, 2025.

\bibitem{dehning2020inferring}
J. Dehning, J. Zierenberg, F. P. Spitzner, M. Wibral, J. P. Neto, M. Wilczek, and V. Priesemann,
``Inferring change points in the spread of COVID-19 reveals the effectiveness of interventions,''
\textit{Science}, vol.~369, no.~6500, pp.~eabb9789, 2020.

\bibitem{mzembegwa2024real}
T. Mzembegwa and C. N. Nyirenda,
``Real-time Pipe Burst Localization in Water Distribution Networks Using Change Point Detection Algorithms,''
in \textit{Proc. 2024 International Conference on Emerging Trends in Networks and Computer Communications (ETNCC)}, pp.~1--8, 2024.

\bibitem{gold2018doubly}
N. Gold, M. G. Frasch, C. L. Herry, B. S. Richardson, and X. Wang,
``A doubly stochastic change point detection algorithm for noisy biological signals,''
\textit{Frontiers in Physiology}, vol.~8, pp.~1112, 2018.

\bibitem{chen2016general}
X. C. Chen, Y. Yao, S. Shi, S. Chatterjee, V. Kumar, and J. H. Faghmous,
``A general framework to increase the robustness of model-based change point detection algorithms to outliers and noise,''
in \textit{Proceedings of the 2016 SIAM International Conference on Data Mining}, pp.~162--170, 2016.

\bibitem{konstantinou2023trend}
A. Konstantinou, D. Chatzakou, O. Theodosiadou, T. Tsikrika, S. Vrochidis, and I. Kompatsiaris,
``Trend detection in crime-related time series with change point detection methods,''
in \textit{International Conference of the Cross-Language Evaluation Forum for European Languages}, pp.~72--84, 2023.


\bibitem{cakmak2024benchmarking}
A. Cakmak, E. Reinertsen, S. Nemati, and G. D. Clifford, ``Benchmarking changepoint detection algorithms on cardiac time series,'' \emph{arXiv preprint arXiv:2404.12408}, 2024.

\bibitem{van2020evaluation}
G. J. J. Van den Burg and C. K. I. Williams, ``An evaluation of change point detection algorithms,'' \emph{arXiv preprint arXiv:2003.06222}, 2020.

\bibitem{zameni2020unsupervised}
M. Zameni, A. Sadri, Z. Ghafoori, M. Moshtaghi, F. D. Salim, C. Leckie, and K. Ramamohanarao, ``Unsupervised online change point detection in high-dimensional time series,'' \emph{Knowledge and Information Systems}, vol. 62, no. 2, pp. 719--750, 2020.

\bibitem{wang2021online}
Z. Wang, X. Lin, A. Mishra, and R. Sriharsha, ``Online changepoint detection on a budget,'' in \emph{2021 International Conference on Data Mining Workshops (ICDMW)}, 2021, pp. 414--420.

\bibitem{theodosiadou2021change}
O. Theodosiadou, K. Pantelidou, N. Bastas, D. Chatzakou, T. Tsikrika, S. Vrochidis, and I. Kompatsiaris, ``Change point detection in terrorism-related online content using deep learning derived indicators,'' \emph{Information}, vol. 12, no. 7, p. 274, 2021.

\bibitem{albertetti2016change}
F. Albertetti, L. Grossrieder, O. Ribaux, and K. Stoffel, ``Change points detection in crime-related time series: an on-line fuzzy approach based on a shape space representation,'' \emph{Applied Soft Computing}, vol. 40, pp. 441--454, 2016.

\bibitem{ref-adwin}
A. Bifet and R. Gavaldà, ``Learning from time-changing data with adaptive windowing,'' in \emph{Proceedings of the 2007 SIAM International Conference on Data Mining}, 2007, pp. 443--448.

\bibitem{ref-page}
E. S. Page, ``Continuous inspection schemes,'' \emph{Biometrika}, vol. 41, no. 1/2, pp. 100--115, 1954.

\bibitem{ref-cusum}
E. S. Page, ``Cumulative sum charts,'' \emph{Technometrics}, vol. 3, no. 1, pp. 1--9, 1961.

\bibitem{ref-bocpd}
R. P. Adams and D. J. C. MacKay, ``Bayesian online changepoint detection,'' \emph{arXiv preprint arXiv:0710.3742}, 2007.

\bibitem{ref-changefinder}
K. Yamanishi and J. Takeuchi, ``A unifying framework for detecting outliers and change points from non-stationary time series data,'' in \emph{Proceedings of the eighth ACM SIGKDD International Conference on Knowledge Discovery and Data Mining}, 2002, pp. 676--681.

\bibitem{ref-rulsif}
Y. Liu, S. Yamada, M. Sugiyama, and N. Murata, ``Change-point detection in time-series data by relative density-ratio estimation,'' \emph{Neural Networks}, vol. 43, pp. 72--83, 2013.

\bibitem{ref-ruptures}
C. Truong, L. Oudre, and N. Vayatis, ``Selective review of offline change point detection methods,'' \emph{Signal Processing}, vol. 167, p. 107299, 2020.

\bibitem{ref-river}
J. Montiel, M. Halford, S. M. Mastelini, G. Bolmier, R. Sourty, R. Vaysse, A. Zouitine, H. M. Gomes, J. Read, T. Abdessalem, and A. Bifet, ``River: machine learning for streaming data in Python,'' \emph{Journal of Machine Learning Research}, vol. 22, no. 110, pp. 1--8, 2021.

\bibitem{ref-ocpdet}
G. van den Burg and C. Williams, ``An evaluation of change point detection algorithms,'' \emph{arXiv preprint arXiv:2003.06222}, 2020.

\bibitem{ref-canary}
M. Chen and D. Tyler, ``Canary: A Python library for online change detection,'' \emph{Software Impacts}, vol. 10, p. 100151, 2021.

% Reference 1
\bibitem{ref-journal}
Author~1, T. The title of the cited article. {\em Journal Abbreviation} {\bf 2008}, {\em 10}, 142--149.
% Reference 2
\bibitem{ref-book1}
Author~2, L. The title of the cited contribution. In {\em The Book Title}; Editor 1, F., Editor 2, A., Eds.; Publishing House: City, Country, 2007; pp. 32--58.
% Reference 3
\bibitem{ref-book2}
Author 1, A.; Author 2, B. \textit{Book Title}, 3rd ed.; Publisher: Publisher Location, Country, 2008; pp. 154--196.
% Reference 4
\bibitem{ref-unpublish}
Author 1, A.B.; Author 2, C. Title of Unpublished Work. \textit{Abbreviated Journal Name} year, \textit{phrase indicating stage of publication (submitted; accepted; in press)}.
% Reference 5
\bibitem{ref-url}
Title of Site. Available online: URL (accessed on Day Month Year).
% Reference 6
\bibitem{ref-proceeding}
Author 1, A.B.; Author 2, C.D.; Author 3, E.F. Title of presentation. In Proceedings of the Name of the Conference, Location of Conference, Country, Date of Conference (Day Month Year); Abstract Number (optional), Pagination (optional).
% Reference 7
\bibitem{ref-thesis}
Author 1, A.B. Title of Thesis. Level of Thesis, Degree-Granting University, Location of University, Date of Completion.
\end{thebibliography}

% If authors have biography, please use the format below
%\section*{Short Biography of Authors}
%\bio
%{\raisebox{-0.35cm}{\includegraphics[width=3.5cm,height=5.3cm,clip,keepaspectratio]{Definitions/author1.pdf}}}
%{\textbf{Firstname Lastname} Biography of first author}
%
%\bio
%{\raisebox{-0.35cm}{\includegraphics[width=3.5cm,height=5.3cm,clip,keepaspectratio]{Definitions/author2.jpg}}}
%{\textbf{Firstname Lastname} Biography of second author}

% For the MDPI journals use author-date citation, please follow the formatting guidelines on http://www.mdpi.com/authors/references
% To cite two works by the same author: \citeauthor{ref-journal-1a} (\citeyear{ref-journal-1a}, \citeyear{ref-journal-1b}). This produces: Whittaker (1967, 1975)
% To cite two works by the same author with specific pages: \citeauthor{ref-journal-3a} (\citeyear{ref-journal-3a}, p. 328; \citeyear{ref-journal-3b}, p.475). This produces: Wong (1999, p. 328; 2000, p. 475)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% for journal Sci
%\reviewreports{\\
%Reviewer 1 comments and authors’ response\\
%Reviewer 2 comments and authors’ response\\
%Reviewer 3 comments and authors’ response
%}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\PublishersNote{}
%\isPreprints{}{% This command is only used for ``preprints''.
\end{adjustwidth}
%} % If the paper is ``preprints'', please uncomment this parenthesis.
\end{document}

